src/main/scala/com/thenewmotion/akka/rabbitmq/ConnectionActor.scala

```diff
@@ -124,7 +124,7 @@ class ConnectionActor(
     log.debug("{} closing broken connection {}", self.path, connection)
     closeIfOpen(connection)
 
-    self ! Connect
+    setTimer(reconnectTimer, Connect, reconnectionDelay / 3, repeat = false)
```

Comment:
I tried to fix this using a different way. The cause of this issue is ChannelActor get two channel in one reconnecting event. 
1. The first time is when ChannelActor get a Shutdown signal. it drops the old channel and send ConnectionActor a ProvideChannel message.
2. After the ConnectionActor re-establish the connection, it creates new channels for each of it's children automatically.

This two channel cause additional close operation, and triggers some race condition. interrupt some `subscribe` or `I/O reading` operation.

I haven't succeed to remove one of channel creation logic. It seems each of them is required under different scenario. 

I couldn't fix this under a global view. So, may be you guys have better idea.